zulipterminal/helper.py

```diff
@@ -678,7 +678,7 @@ def process_media(controller: Any, media_link: str) -> None:
     elif MACOS:
         command = 'open'
 
-    open_media(controller, command, media_path)
+    controller.show_media_confirmation_popup(open_media, command, media_path)
```

Comment:
I'm guessing you're trying to synchronize the asynch download terminating?

Reply:
I'm not sure if I fully understood your comment, but I think the callback would be working fine in any case?